MyStore/src/MyStore.cpp

```diff
@@ -3,35 +3,61 @@
 #include <chrono>
 #include <windows.h>
 #include <filesystem>
+#include <thread>
+#include <mutex>
 
 using namespace std::chrono;
 
 // This variable is needed to avoid confusion with relative paths in different modules
 // Example of usage:
 // std::string filePath = gProjectPath + "\\data\\Data.json"
+
+void GenerateProducts(StorageData& stor, const int& period, bool& flag);
+void ExitCondition(bool& flag);
 const std::string gProjectPath = std::filesystem::current_path().generic_string();
+std::mutex mtx;
 
 int main(int argc, char* argv[])
 {
 	StorageData storage;
+	bool flag_thread = false;

```

Comment:
It's better to use atomic<bool> as a synchronization primitive in such cases.

Reply:
Okay, great idea!